packages/shared/lib/core/wallet/utils/isValidIRC30.ts

```diff
@@ -0,0 +1,10 @@
+import { NativeTokenBalance } from '@iota/wallet/out/types'
+import { IPersistedAsset } from '../interfaces/persisted-asset.interface'
+
+export function isValidIRC30(asset: IPersistedAsset, balance: NativeTokenBalance): boolean {
+    const hasValidDecimals = Number(asset.metadata.decimals) < Number.MAX_SAFE_INTEGER
+    const hasValidTotal = Number(balance.total) < Number.MAX_SAFE_INTEGER
+    const hasValidAvailable = Number(balance.available) < Number.MAX_SAFE_INTEGER
+
+    return hasValidDecimals && hasValidTotal && hasValidAvailable
```

Comment:
This is a limitation of JS, not of IRC30, so the name is incorrect. Possibly we should look into how to handle larger values by using BigInt or Big library in Javascript. Firefly should be able to handle these values.